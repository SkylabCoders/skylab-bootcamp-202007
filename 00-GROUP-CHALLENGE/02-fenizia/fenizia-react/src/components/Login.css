.login-main-container {
  background-color: #fff;
  height: 80vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.login-main-container__wrap {
  margin: auto;
  width: 40vw;
  height: 60vh;
  border: solid 4px #333;
  border-radius: 2%;
  text-align: center;
}
.login-wrap__form__logo {
  width: 20%;
  filter: grayscale(100%);
}
.login-wrap__form {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  height: 100%;
}
.login-wrap__form__btn > button {
  border-style: none;
  padding: 2vh;
  color: black;
  background-color: #ffcc00;
  border: 2px transparent solid;
  width: 10vw;
  min-width: 70px;
}
.login-wrap__form__btn > button:hover {
  border: 2px #333 solid;
}
.login-wrap__form__btn > img {
  width: 40%;
  margin: 2.2vw;
  border: 2px solid transparent;
}
.login-wrap__form__btn > img:hover {
  border: 2px solid #ffcc00;
  border-radius: 9%;
}
.logout-btn {
  background-color: #ffcc00;
  margin-left: 80vw;
  margin-top: 2vh;
  padding: 0.6vw;
  color: #fff;
  border: 2px solid transparent;
}
.logout-btn:hover {
  color: #333;
  border: 2px solid #333;
}
.input__search {
  margin: auto;
}
/* QUERIES */

@media screen and (max-width: 599px) {
  .login-main-container__wrap {
    width: 100vw;
    height: 80vh;
    border: solid 4px transparent;
  }
}
