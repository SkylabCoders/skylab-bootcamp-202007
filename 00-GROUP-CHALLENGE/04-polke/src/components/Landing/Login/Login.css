/*
** FORM INSIDE CONTAINER
*/

.Login__form {
	background-color: var(--form-color-background);
	width: 80%;
	opacity: 0.8;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	padding-bottom: 10px;
	box-shadow: 2px 2px #888888;
	text-align: center;
}

/*
** FORM ELEMENTS
*/

.Login__form__login-title--colored {
	color: var(--form-color-headers);
	padding-bottom: 5px;
}

.Login__login-with-button {
	width: 100% !important;
}
.Login__login-with-button button {
	width: 80% !important;
	height: 60px;
}
.Login__form__container-input > a {
	color: var(--font-color-input);
}
.Login__form__container-input {
	width: 80% !important;
}

.Login__form__anchor--colored {
	color: var(--form-color-headers);
	font-size: bold;
	font-weight: bolder;
	cursor: pointer;
}

.Login__form__last-button {
	width: 100% !important;
	height: 40px;
	margin-top: 12px;
}

.form__button {
	background-color: var(--bg-color-button);
	border-radius: 5px;
	border: none;
	color: whitesmoke;
	cursor: pointer;
}

.form__button a {
	color: white;
}

.login__log-anonym {
	cursor: pointer;
}

.anchor:hover {
	color: #2d8ff9 !important;
}

/*
** MEDIA QUERIES
*/

@media only screen and (min-width: 600px) {
	.Login__login-with-button button {
		width: 70% !important;
		height: 60px;
	}
	.Login__form__container-input {
		width: 70% !important;
	}
}

@media only screen and (min-width: 1024px) {
	.Login__form {
		min-width: 375px !important;
		max-width: 450px !important;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		text-align: center;
	}

	.Login__form button {
		width: 300px !important;
	}
	.Login__form__container-input {
		width: 300px !important;
	}
}
