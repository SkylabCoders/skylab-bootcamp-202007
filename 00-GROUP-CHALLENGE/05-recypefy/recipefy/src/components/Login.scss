.login__title {
	padding: 3px;
	border-top: 2px solid rgba(128, 128, 128, 0.658);
	border-bottom: 2px solid rgba(128, 128, 128, 0.658);
}

.login__box {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	background-color: #65dfc3;
	min-width: 286px;
	padding: 16px;
	border-radius: 10px;
	margin-bottom: 80px;
}

form {
	text-align: center;
}

.form__input {
	margin-top: 5px;
	margin-bottom: 15px;
	height: 30px;
	min-width: 250px;
	width: 20vw;
	padding: 2px;
	border-radius: 4px;
	background-color: rgba(128, 128, 128, 0.514);
	text-align: center;
	font-size: 18px;
	border: 1px solid #65dfc3;
}

.login__button {
	background: none;
	background-color: #b7ef9d;
	border: 1px solid black;
	padding: 8px 16px;
	font: inherit;
	cursor: pointer;
	outline: inherit;
	border-radius: 4px;
	margin-bottom: 10px;
}

.login__container {
	width: 100%;
	margin-top: 80px;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;

	p {
		border-top: 1px solid black;
		border-bottom: 1px solid black;
		margin: 10px;
		width: 10vw;
		text-align: center;
	}

	h2 {
		margin: 8px;
	}

	a {
		padding: 12px;
		margin: 5px;
		background-color: rgba(156, 149, 149, 0.24);
		border-radius: 4px;
		min-width: 80px;
		width: 12vw;
		text-align: center;
		cursor: pointer;
	}

	a:hover {
		background-color: rgba(156, 149, 149, 0.651);
		color: honeydew;
	}
}

.register__form {
	background-color: rgba(128, 128, 128, 0.514);
	border-radius: 8px;
	padding: 10px;
}

.info__title {
	margin-bottom: 10px;
}

.api__link:hover {
	color: honeydew;
}

.login__logo--size {
	width: 50%;
}
