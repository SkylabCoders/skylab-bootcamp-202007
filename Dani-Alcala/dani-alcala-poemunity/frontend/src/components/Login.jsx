import React from "react";
import { useAuth0 } from "@auth0/auth0-react";
import "./Header.scss";

const Login = () => {
  const { loginWithRedirect } = useAuth0();

  return <button className="header__login" onClick={() => loginWithRedirect()}></button>;
};

export default Login;