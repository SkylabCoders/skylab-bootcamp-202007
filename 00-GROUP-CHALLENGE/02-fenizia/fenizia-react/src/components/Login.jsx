import React, { useState, useEffect } from "react";
import "./Login.css";
import { login, logout, loginWithGoogle } from "../actions/authActions";
import authStore from "../stores/authStore";
import Profile from "./Profile";
import TextInput from "./TextInput";

function Login() {
  const [isLogged, setIsLogged] = useState(authStore.isLogged());
  const [user, setUser] = useState(authStore.getUserProfile());
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  useEffect(() => {
    authStore.addChangeListener(onAuthChange);

    return () => authStore.removeChangeListener(onAuthChange);
  }, [isLogged, user]);
  function onFieldChange(value, setValue) {
    setValue(value);
  }
  function onAuthChange() {
    setIsLogged(authStore.isLogged());
    setUser(authStore.getUserProfile());
  }

  return (
    <>
      {!isLogged && (
        <>
          <div className="login-main-container">
            <div className="login-main-container__wrap">
              <div className="login-wrap__form">
                <span>
                  <img
                    src="https://image.flaticon.com/icons/svg/1000/1000946.svg"
                    className="login-wrap__form__logo"
                    alt="login"
                  />
                </span>
                <span>Log in</span>

                <div data-validate="Enter username">
                  <TextInput
                    type="text"
                    name="username"
                    placeholder="Username"
                    value={email}
                    onChange={(event) =>
                      onFieldChange(event.target.value, setEmail)
                    }
                  />
                  <span data-placeholder=""></span>
                </div>

                <div data-validate="Enter password">
                  <TextInput
                    type="password"
                    textContentType={"password"}
                    name="pass"
                    placeholder="Password"
                    value={password}
                    onChange={(event) =>
                      onFieldChange(event.target.value, setPassword)
                    }
                  />
                  <span data-placeholder=""></span>
                </div>

                <div className="login-wrap__form__btn">
                  <button onClick={() => login(email, password)}>Login</button>
                  <a href="/register">
                    <p>Sign in</p>
                  </a>
                  <img
                    src="https://icon-library.com/images/login-with-google-icon/login-with-google-icon-3.jpg"
                    onClick={() => loginWithGoogle(email, password)}
                    alt="Log in with google"
                  ></img>
                </div>
              </div>
            </div>
          </div>
        </>
      )}
      {isLogged && (
        <>
          <button className="logout-btn" onClick={() => logout()}>
            Logout
          </button>
          <Profile />
        </>
      )}
    </>
  );
}

export default Login;
