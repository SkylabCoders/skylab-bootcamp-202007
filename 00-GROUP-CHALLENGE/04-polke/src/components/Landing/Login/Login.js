import React, { useState, useEffect } from 'react';
import { FormControl, FormGroup } from 'react-bootstrap';
import {
	login,
	loginGoogle,
	loginAnonymously
} from '../../../actions/loginActions';
import landingStore from '../../../stores/landingStore';
import '../../../shared/generalStyles.css';
import './Login.css';
import PopUpBox from '../PopUpBox/PopUpBox';
import { withRouter } from 'react-router-dom';
import githubApiConst from '../../../shared/githubApiConst';

export default withRouter(function ({ history }) {
	const [isLogged, setIsLogged] = useState(false);
	const [user, setUser] = useState(null);

	const USERDETAIL_PATH = './userDetail';

	useEffect(() => {
		landingStore.addChangeListener(onAuthChange);
		user ? history.push(USERDETAIL_PATH) : history.push('');
		return () => landingStore.removeChangeListener(onAuthChange);
	}, [isLogged, user]);

	function onAuthChange() {
		setIsLogged(landingStore.isLogged());
		setUser(landingStore.getUserProfile());
	}

	function handleLoginClick(event) {
		event.preventDefault();
		const email = document.getElementsByClassName('input-email')[0].value;
		const password = document.getElementsByClassName('input-password')[0].value;
		login(email, password);
	}

	return (
		<>
			<form className="Login__form">
				<h2 className="Login__form__login-title--colored">Log In</h2>
				<FormGroup
					className="Login__login-with-button"
					controlId="username"
					bsSize="large"
				>
					<button
						className="form__button"
						onClick={(event) => {
							event.preventDefault();
							loginGoogle();
						}}
					>
						Google
					</button>
				</FormGroup>
				<FormGroup
					className="Login__login-with-button "
					controlId="password"
					bsSize="large"
				>
					<button className="form__button">
						<a
							href={`https://github.com/login/oauth/authorize?client_id=${githubApiConst.GITHUB_CLIENTID}&scope=${githubApiConst.GITHUB_SCOPE}&login`}
						>
							GitHub
						</a>
					</button>
				</FormGroup>
				<FormGroup
					className="Login__form__container-input"
					controlId="input"
					bsSize="large"
				>
					<FormControl
						className="Login__form__input input-email"
						placeholder="&#128100;"
					/>
					<FormControl
						className="Login__form__input input-password"
						type="password"
						placeholder="&#128272;"
					/>

					<button
						className="Login__form__last-button form__button"
						onClick={(event) => {
							handleLoginClick(event);
						}}
					>
						Enter
					</button>
					<a
						className="Login__form__anchor--colored anchor Login-anchor"
						onClick={(event) => {
							event.preventDefault();
							document.getElementsByClassName('pop-up-modal')[0].style.display =
								'block';
						}}
					>
						Register
					</a>
					{'   |   '}
					<a
						className="login__log-anonym anchor Login-anchor"
						onClick={(event) => {
							event.preventDefault();
							loginAnonymously();
						}}
					>
						Log In Anonymously
					</a>
				</FormGroup>
			</form>
			<PopUpBox />
		</>
	);
});
