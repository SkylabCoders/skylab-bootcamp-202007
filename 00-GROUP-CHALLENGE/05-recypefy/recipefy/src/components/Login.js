import React, { useState, useEffect } from 'react';
import { login, logout, googleLogin } from "../actions/AuthActions";
import authStore from "../stores/AuthStore";
import './Login.css';


function Login(){
   
    const [user, setUser] = useState(authStore.getUserProfile());
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('');
    // function validateForm(){ 
    //     return email.length > 0 && password.length > 0
    // }
    

    const [isLogged, setIsLogged] = useState(authStore.isLogged())

    useEffect(()=> {
        authStore.addChangeListener(onAuthChange);
       
        return () => authStore.removeChangeListener(onAuthChange);
    }, [isLogged, user]);

    function onAuthChange(){
        setIsLogged(authStore.isLogged());
        setUser(authStore.getUserProfile());
    }

    

    return(
        <div className="login__container">
            <div className="login__box">
                <form>
                    <label for="email">Email</label><br/>
                    <input autoFocus type="email" name="email" className="form__input" value={email} onChange={(e) => { setEmail(e.target.value); console.log(email)}}/><br/>
                    <label>Password</label><br/>
                    <input type="password" className="form__input" value={password} onChange={e => setPassword(e.target.value)}/><br/>
                
                    {!isLogged &&(
                        
                        <button onClick={(event) => { event.preventDefault(); login(email, password)}}>Login</button>
                        )}
                    
                    {isLogged && (
                   
                    <button onClick = {(event) => { event.preventDefault();logout()}}>Logout {user && user.user.email}</button>
                    )}

                </form>
                

                <p>or</p>
                <h2>REGISTER</h2>
                <a onClick={googleLogin}>GOOGLE</a>
                <a>EMAIL</a>
                <a>GUEST</a>
            </div>
        </div>

    );
}

export default Login;