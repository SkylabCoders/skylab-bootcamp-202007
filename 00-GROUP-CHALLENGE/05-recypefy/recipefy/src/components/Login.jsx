import React, { useState, useEffect } from 'react';
import { login, logout, googleLogin, anonymousLogin, createUser } from "../actions/AuthActions";
import authStore from "../stores/AuthStore";
import './Login.scss';


function Login() {
    const [user, setUser] = useState(authStore.getUserProfile());
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('');
    const [confirmPassword, setConfirmPassword] = useState('')
    const [showRegister, setRegister] = useState(false);
    const [showLogin, setLogForm] = useState(false);
    const [isLogged, setIsLogged] = useState(authStore.isLogged());


    useEffect(() => {
        authStore.addChangeListener(onAuthChange);

        return () => authStore.removeChangeListener(onAuthChange);
    }, [isLogged, user]);

    function onAuthChange() {
        setIsLogged(authStore.isLogged());
        setUser(authStore.getUserProfile());
    }

    function displayLogin() {
        if (!showLogin) {
            setLogForm(true);
        } else {
            setLogForm(false);
        }
    }

    function displayRegister() {
        if (!showRegister) {
            setRegister(true);
        } else {
            setRegister(false);
        }
    }

    function validateForm() {
        return email.length > 0 && password.length > 0
    }


    function validateRegister() {
        if (password === confirmPassword) {
            createUser(email, password);
            alert('You registered successfully!!')
        } else {
            alert('Both passwords has to match.')
        }
    }

    return (
        <div className="login__container">
            <div className="login__box">
                <img className="login__logo--size" src='https://image.flaticon.com/icons/svg/770/770906.svg' />
                <h2 className="login__title">Login with</h2>
                <form>

                    {showLogin && (<>
                        <label className='hide__element' for="email">Email</label><br />
                        <input autoFocus type="email" name="email" className="form__input hide__element" value={email} onChange={(e) => { setEmail(e.target.value) }} /><br />
                        <label className='hide__element' >Password</label><br />
                        <input type="password" className="form__input hide__element" value={password} onChange={e => setPassword(e.target.value)} /><br />


                        {!isLogged && (

                            <button className="login__button hide__element" disabled={!validateForm()} onClick={(event) => { event.preventDefault(); login(email, password) }}>Login</button>
                        )}
                    </>)}

                    {isLogged && (

                        <button className="login__button" onClick={(event) => { event.preventDefault(); logout() }}>Logout {(user && user.user.email) || "Anonymous"}</button>
                    )}

                </form>

                <a onClick={displayLogin}>EMAIL</a>
                <a onClick={googleLogin}>GOOGLE</a>
                <a onClick={anonymousLogin}>ANON</a>
                <h2 className="login__title">OR</h2>
                <a onClick={displayRegister}>REGISTER</a>

                {showRegister && (<>
                    <form className="register__form">

                        <label for="email">Email</label><br />
                        <input autoFocus type="email" name="email" className="form__input" value={email} onChange={(e) => { setEmail(e.target.value) }} /><br />
                        <label>Password</label><br />
                        <input type="password" className="form__input" value={password} onChange={e => setPassword(e.target.value)} /><br />
                        <label>Confirm Password</label><br />
                        <input type="password" className="form__input" value={confirmPassword} onChange={e => setConfirmPassword(e.target.value)} /><br />

                        <button className="login__button" onClick={(event) => { event.preventDefault(); validateRegister() }}>Register</button>
                    </form>
                </>
                )}


            </div>
        </div>

    );
}

export default Login;