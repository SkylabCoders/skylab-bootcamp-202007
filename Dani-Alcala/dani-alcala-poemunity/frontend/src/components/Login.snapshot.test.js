import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';
import { BrowserRouter } from 'react-router-dom';

function renderLogin(arg) {
	const defaultProps = {
		match: {
			params: {}
		}
	};

	const props = { ...defaultProps, ...arg };
	return renderer.create(
		<BrowserRouter>
			<Login {...props} />
		</BrowserRouter>
	);
}

describe('Login', () => {
	let LoginTree;

	beforeEach(async () => {
		LoginTree = renderLogin();
	});

	it('should match without id', async () => {
		expect(LoginTree).toMatchSnapshot();
    });
})