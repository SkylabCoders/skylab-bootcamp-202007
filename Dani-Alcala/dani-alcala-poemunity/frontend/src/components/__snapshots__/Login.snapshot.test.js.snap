// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match without id 1`] = `
<button
  className="header__login"
  onClick={[Function]}
/>
`;
